### ISSUE TITLE
401 error cause basic auth popup to show multiple times

### ISSUE THREAD

**Issue by marco-giovannini-73team** [kubecost/cost-analyzer-helm-chart#1515]
**Describe the bug**

I am using gcr.io/kubecost1/cost-model:prod-1.94.3

When using an nginx ingress with basic authentication and a different path then the / to expose kubecost authentication popup keep showing.

**To Reproduce**

1. Expose Kubecost UI using an nginx ingress with basic authentication on a different path:

Example

```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    nginx.ingress.kubernetes.io/auth-type: basic
    nginx.ingress.kubernetes.io/auth-secret: basic-auth
    kubernetes.io/ingress.class: xxx-nginx
    meta.helm.sh/release-name: logmon
    meta.helm.sh/release-namespace: xxx-logmon
    nginx.ingress.kubernetes.io/configuration-snippet: |
      rewrite ^(/kubecost)$ $1/ permanent;
    nginx.ingress.kubernetes.io/rewrite-target: /$2
  labels:
    app: cost-analyzer
    app.kubernetes.io/instance: logmon
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: cost-analyzer
    helm.sh/chart: cost-analyzer-1.94.3
  name: logmon-cost-analyzer
  namespace: xxx-logmon
spec:
  rules:
  - host: xxx.eastus.cloudapp.azure.com
    http:
      paths:
      - backend:
          service:
            name: logmon-cost-analyzer
            port:
              name: tcp-frontend
        path: /kubecost(/|$)(.*)
        pathType: ImplementationSpecific
  tls:
  - hosts:
    - xxx.eastus.cloudapp.azure.com
    secretName: xxx-tls-secret
```

2. Login first time and navigate in another sections part of the applications. 
    The authentication popup will show over and over again.

**Expected behavior**
Login only once should be required

**Screenshots**

![Screenshot 2022-07-05 at 16 27 32](https://user-images.githubusercontent.com/95924236/177361404-a0b9af6d-f3b5-41a1-9d9b-664f13b721eb.png)

urls of favicon and azure_logo_square (in the diagnostic page) don't get rewritten properly causing 401 errors and causing the basic authentication popup again.

<img width="1210" alt="Screenshot 2022-07-05 at 14 48 09" src="https://user-images.githubusercontent.com/95924236/177361791-ab6fc932-51d8-4105-bcc8-1c2606603104.png">

![Screenshot 2022-07-05 at 16 45 40](https://user-images.githubusercontent.com/95924236/177362235-afa07901-744c-4e3d-96d5-4a5a8adbc41b.png)
![Screenshot 2022-07-05 at 16 46 11](https://user-images.githubusercontent.com/95924236/177362247-337ef236-eb2b-4e37-86d5-cc7671814983.png)

gz#2099<br><br><i>(related to <a href='https://kubecost.zendesk.com/agent/tickets/2099'>Zendesk ticket #2099</a>)</i>

**Comment by Adam-Stack-PM** [comment]
@kaelanspatel Can you triage this issue and label it P0, P1 or P2?

**Comment by kaelanspatel** [comment]
Sure Adam!

@marco-giovannini-73team just to clarify, if you set the path to the one used for Kubecost, this works as intended i.e. only shows once?

**Comment by marco-giovannini-73team** [comment]
@kaelanspatel  I wasn't able to test in our environment  as we have other applications in / so cannot configure Kubecost ingress on the / path but I will be able to test on my local if needed . This is not the first time I encounter this issue with the basic auth popup when the application trying to load the page encounter some 40X error in particular with Chrome.

**Comment by marco-giovannini-73team** [comment]
I did few more testing  in another environment.

with the ingress configured on the root / path no 4XX error is showed and one authenticated  is required no other popup is showed

<img width="1521" alt="Screenshot 2022-07-06 at 17 41 43" src="https://user-images.githubusercontent.com/95924236/177594934-a7c8ddce-68e4-4444-a178-92045240f1b3.png">

with the ingress configured on the /kubecost path a 404 error is showed and one authenticated  is required no other popup is showed

<img width="1515" alt="Screenshot 2022-07-06 at 17 46 33" src="https://user-images.githubusercontent.com/95924236/177595220-52654c10-fdcf-4cfb-a0eb-78ff35454220.png">

The only diffrerent between the two cases is on the first I get 401 error in this second case I get 404

**Comment by marco-giovannini-73team** [comment]
On the first cluster with the issue we added the following configuration to the ingress an we stopped receiving the basic auth popup all the time.

```
nginx.ingress.kubernetes.io/server-snippet: |                                                                                                                                                    │
│       location ~* "^/favicon.ico" {                                                                                                                                                                  │
│           deny all;                                                                                                                                                                                  │
│           return 200;                                                                                                                                                                                │
│         }                                                                                                                                                                                            │
│       location ~* "^/images/azure_logo_square.png" {                                                                                                                                                 │
│           deny all;                                                                                                                                                                                  │
│           return 200;                                                                                                                                                                                │
│         }
```

So the issue is related to the loading errors but in it is shows only on the case of 401 not 404.

**Comment by kaelanspatel** [comment]
Thanks for the additional context @marco-giovannini-73team, we'll get this looked at.

**Comment by Adam-Stack-PM** [comment]
@nealormsbee Could you get a priority Assigned to this (P1,2,3) and add a category?

**Comment by nealormsbee** [comment]
Sure thing. I will do some additional testing here, but given that the issues are wrong URLs for `favicon` and the Azure logo, this may be resolved in 1.95.0. I know that at least the `favicon.ico` will now route correctly.

Labeling `P1` because this is pretty serious, but @marco-giovannini-73team has supplied a successful workaround.

**Comment by nealormsbee** [comment]
Based on my additional testing, this issue should be resolved in `1.95.0` and `1.96.0` when we release it ~next week.

@marco-giovannini-73team could I persuade you to upgrade, remove the ingress workaround, and let us know whether the issue is resolved?

**Comment by nealormsbee** [comment]
Closing for now. If it turns out that I've missed something, please feel free to re-open and @ me
